lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -205,135 +210,223 @@ def _on_list_backups_action(self, event: ActionEvent) -> None:
         else:
             event.fail("Failed: invalid output format, must be either json or table")
 
-    def _restore_and_try_close_indices_if_needed(
-        self, backup_id: int
-    ) -> Tuple[Dict[str, Any], Set[str]]:
-        """Restores and processes any exception related to running indices.
+    def _request_indices_to_close(self, indices: Set[str]) -> Dict[str, Any]:
+        """Requests indices to be closed and return their status."""
+        resp = self._request(
+            "POST",
+            f"{','.join(indices)}/_close",
+            payload={
+                "ignore_unavailable": "true",
+            },
+        )
+        if not resp or not resp.get("acknowledged", False):
+            return {"acknowledged": False}
```

Comment:
@Mehdi-Bendriss yes, it changes the "shards_ack" field:
```
$ curl -XPOST -k -u admin:PWD https://10.206.26.149:9200/series_index,test_backup_index,.plugins-ml-config/_close
{"acknowledged":true,"shards_acknowledged":true,"indices":{"series_index":{"closed":true},"test_backup_index":{"closed":true},".plugins-ml-config":{"closed":true}}}
$ curl -XPOST -k -u admin:PWD https://10.206.26.149:9200/series_index,test_backup_index,.plugins-ml-config/_close
{"acknowledged":true,"shards_acknowledged":false,"indices":{}}
```

I will make it a binary method.

Reply:
@Mehdi-Bendriss I think that `_close_indices` should return True if we acknowledged. Tbh, the risk of having a missing index is not really high, because that method is only called: `_close_indices_if_needed` and we do a check between backup indices, available indices currently, indices that are already closed and indices that should be excluded of a restore.

(I also renamed _close_indices to _call_close_index_api, as the names were getting confusing :)